simpleflow/swf/task.py

```diff
@@ -178,6 +203,32 @@ def schedule(self, domain, task_list=None, **kwargs):
 
         return [decision]
 
+    @classmethod
+    def get_workflow_type(cls, domain, name, version):
+        # type: (swf.models.Domain, str, str) -> swf.models.WorkflowType
+        """
+        Cache known WorkflowType's to remove useless latency.
+        :param domain:
+        :type domain:
+        :param name:
+        :type name:
+        :param version:
+        :type version:
+        :return:
+        :rtype:
+        """
+        key = (domain.name, name, version)
+        model = cls.cached_models.get(key)
+        if model:
+            return model
+        model = swf.models.WorkflowType(
+            domain,
+            name,
+            version=version,
+        )
+        cls.cached_models[key] = model
+        return model
```

Comment:
Er, I usually do it quite like this... Dunno why the reverse logic here. Will change it!